sdk/tables/Microsoft.Azure.WebJobs.Extensions.Tables/src/PocoEntityArgumentBinding.cs

```diff
@@ -2,34 +2,45 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
 using System.Threading.Tasks;
+using Azure;
+using Azure.Data.Tables;
 using Microsoft.Azure.WebJobs.Host.Bindings;
-using Microsoft.Azure.Cosmos.Table;
 
 namespace Microsoft.Azure.WebJobs.Extensions.Tables
 {
     internal class PocoEntityArgumentBinding<TElement> : IArgumentBinding<TableEntityContext>
         where TElement : new()
     {
-        private static readonly IConverter<ITableEntity, TElement> Converter =
-            TableEntityToPocoConverter<TElement>.Create();
-
         public Type ValueType => typeof(TElement);
 
         public async Task<IValueProvider> BindAsync(TableEntityContext value, ValueBindingContext context)
         {
             var table = value.Table;
-            var retrieve = table.CreateRetrieveOperation<DynamicTableEntity>(
-                value.PartitionKey, value.RowKey);
-            TableResult result = await table.ExecuteAsync(retrieve, context.CancellationToken).ConfigureAwait(false);
-            DynamicTableEntity entity = (DynamicTableEntity)result.Result;
-            if (entity == null)
+            TableEntity entity;
+            try
+            {
+                var result = await table.GetEntityAsync<TableEntity>(
+                    value.PartitionKey, value.RowKey).ConfigureAwait(false);
+                entity = result.Value;
+            }
+            catch (RequestFailedException e) when (e.Status == 404 && e.ErrorCode == "TableNotFound")
```

Comment:
Do we want to filter on this error code? `GetEntity` will also throw 404 if the entity doesn't exist.

Reply:
Yep, fixed and added a test.